test/unit/transformer-oracle.spec.ts

```diff
@@ -0,0 +1,61 @@
+import chai, { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { constants } from 'ethers';
+import { behaviours } from '@utils';
+import { then, when } from '@utils/bdd';
+import { ITransformerRegistry, ITokenPriceOracle, TransformerOracle, TransformerOracle__factory } from '@typechained';
+import { snapshot } from '@utils/evm';
+import { smock, FakeContract } from '@defi-wonderland/smock';
+
+chai.use(smock.matchers);
+
+describe('TransformerOracle', () => {
+  let transformerOracleFactory: TransformerOracle__factory;
+  let transformerOracle: TransformerOracle;
+  let underlyingOracle: FakeContract<ITokenPriceOracle>;
+  let registry: FakeContract<ITransformerRegistry>;
+  let snapshotId: string;
+
+  before('Setup accounts and contracts', async () => {
+    transformerOracleFactory = await ethers.getContractFactory('solidity/contracts/TransformerOracle.sol:TransformerOracle');
+    registry = await smock.fake<ITransformerRegistry>('ITokenPriceOracle');
```

Comment:
Wait, faking a `ITokenPriceOracle` seems off? Probably this was fixed down the line tho.

Reply:
Indeed, it was fixed in another PR 😊